skbio/io/format/genbank.py

```diff
@@ -0,0 +1,468 @@
+from pprint import pprint
+
+genbank = create_format('genbank')
+
+# date format in locus line of genbank record
+_TIME_FORMAT = '%d-%b-%Y'
+# This list is ordered
+# used to read and write genbank file.
+_HEADERS = ['LOCUS',
+            'DEFINITION',
+            'ACCESSION',
+            'VERSION',
+            'DBSOURCE',
+            'DBLINK',
+            'KEYWORDS',
+            'SOURCE',
+            'REFERENCE',
+            'COMMENT',
+            'FEATURES',
+            'ORIGIN']
+
+
+@genbank.sniffer()
+def _genbank_sniffer(fh):
+    # check the 1st line starts with 'LOCUS' and the last line ends with '//'
+    if _too_many_blanks(fh, 5):
+        return False, {}
+    try:
+        line = next(_line_generator(fh, skip_blanks=True))
+    except StopIteration:
+        return False, {}
+
+    if line.startswith(_HEADERS[0]):
+        for line in _line_generator(fh, skip_blanks=True):
+            if line == '//':
+                return True, {}
+        return False, {}
+    else:
+        return False, {}
+
+
+@genbank.reader(None)
+def _genbank_to_generator(fh, constructor=None, **kwargs):
+    for record in _parse_genbanks(fh):
+        yield _construct(record, constructor, **kwargs)
+
+
+@genbank.reader(Sequence)
+def _genbank_to_biological_sequence(fh, rec_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), rec_num)
+    return _construct(record, Sequence, **kwargs)
+
+
+@genbank.reader(DNA)
+def _genbank_to_dna(fh, rec_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), rec_num)
+    return _construct(record, DNA, **kwargs)
+
+
+@genbank.reader(RNA)
+def _genbank_to_rna(fh, rec_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), rec_num)
+    return _construct(record, DNA, **kwargs).transcribe()
+
+
+@genbank.reader(Protein)
+def _genbank_to_protein(fh, rec_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), rec_num)
+    return _construct(record, Protein, **kwargs)
+
+
+@genbank.writer(None)
+def _sequences_to_genbank(fh):
+    pass
+
+
+def _construct(record, constructor=None, **kwargs):
+    seq, md, pmd = record
+    if constructor is None:
+        unit = md['LOCUS']['unit']
+        if unit == 'bp':
+            # RNA mol type has T instead of U;
+            # so still read in as DNA
+            constructor_ = DNA
+        elif unit == 'aa':
+            constructor_ = Protein
+        else:
+            constructor_ = Sequence
+    else:
+        constructor_ = constructor
+    return constructor_(seq, metadata=md, positional_metadata=pmd, **kwargs)
+
+
+def _parse_genbanks(fh):
+    data_chunks = []
+    for line in _line_generator(fh, skip_blanks=True, strip=False):
+        if line.startswith('//'):
+            yield _parse_single_genbank(data_chunks)
+            data_chunks = []
+        else:
+            data_chunks.append(line)
+
+
+def _parse_single_genbank(chunks):
+    metadata = dict()
```

Comment:
it is pure stylistic, or because `dict` is a little bit slower for `dict` lookup?